scripts/Muon/GUI/Common/plot_widget/base_pane/base_pane_view.py

```diff
@@ -0,0 +1,187 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from qtpy import QtWidgets
+
+from mantidqt.utils.qt import load_ui
+#from Muon.GUI.Common.plot_widget.plot_widget_view_interface import PlotWidgetViewInterface
+import Muon.GUI.Common.message_box as message_box
+
+ui_plotting_view, _ = load_ui(__file__, "base_pane_view.ui")
+
+
+class BasePaneView(QtWidgets.QWidget, ui_plotting_view):
+
+    @staticmethod
+    def warning_popup(message):
+        message_box.warning(str(message))
+
+    def __init__(self, parent=None):
+        super().__init__(parent=parent)
+        self.setupUi(self)
+        self.setMinimumSize(600,600)
+        #self.setEnabled(False)
+
+    def show_plot_diff(self):
+        self.plot_diff_checkbox.setVisible(True)
+
+    def hide_plot_diff(self):
+        self.plot_diff_checkbox.setVisible(False)
+
+    def setup_plot_type_options(self, options):
+        """
+        Setup the options which are displayed in the plot type combo box
+        """
+        self.plot_type_combo.blockSignals(True)
+        self.plot_type_combo.clear()
+        for option in options:
```

Comment:
I think we can avoid these for loops by using `QComboBox.addItems` which takes a list of items